source/language/types.rst

```diff
@@ -135,8 +124,63 @@ circuits. Physical qubits must not be declared and they are, as all the qubits,
    qubit γ;
    // Declare a qubit register with 20 qubits
    qubit[20] qubit_array;
+
+.. _physical-qubits:
+
+Physical Qubits
+~~~~~~~~~~~~~~~
+
+Physical qubits refer to particular hardware qubits. Therefore, they are only fully defined with
+respect to a target device that has a published device topology. The hardware provider determines
+the integer-labels associated with each qubit within the device's topology.
+
+While virtual qubits can be named, hardware qubits are referenced by the syntax ``$[NUM]``, where
+``NUM`` is a non-negative integer. Any integer included in the published device topology is a valid
+physical qubit identifier. Note that this implies that physical qubit identifier indices may be
+non-consecutive, depending on the device.
+
+Like virtual qubits, physical qubits are global variables, but unlike virtual qubits, they must
+not be declared.
+
+These qubit types are always used for ``defcal`` blocks. Calibrations are valid only for a
+particular set of physical qubits. See also :ref:`pulse gates <pulse-gates>`.
```

Comment:
*These qubit types* (which is copied from the previous text) should be *This qubit type*. But I think it would be better to make this even more precise. It's not clear to me what the sentence means, in particular *used for*. Does it mean that the qubit parameters in a `defcal` declaration must be physical qubits? Or that the qubit arguments in a gate call that dispatches to a `defcal` "method" must be physical qubits? And/or that the qubits appearing in the body of a `defcal` declaration must be physical qubits?

I guess that it would be easier and more automatic to write precise and concise statements if we had an unambiguous language to talk about the language elements. For example, using the conventional programming language distinction between *parameter* and *argument*. The spec makes this distinction sometimes, but not always. Pursuing this observation belongs in another issue, but I think we could still make an improvement to this text.